app/src/test/kotlin/io/eugenethedev/taigamobile/repositories/TasksRepositoryTest.kt

```diff
@@ -0,0 +1,945 @@
+package io.eugenethedev.taigamobile.repositories
+
+import io.eugenethedev.taigamobile.data.repositories.SprintsRepository
+import io.eugenethedev.taigamobile.data.repositories.TasksRepository
+import io.eugenethedev.taigamobile.data.repositories.UsersRepository
+import io.eugenethedev.taigamobile.domain.entities.CommonTaskType
+import io.eugenethedev.taigamobile.domain.entities.FiltersData
+import io.eugenethedev.taigamobile.domain.entities.UsersFilter
+import io.eugenethedev.taigamobile.domain.repositories.ISprintsRepository
+import io.eugenethedev.taigamobile.domain.repositories.ITasksRepository
+import io.eugenethedev.taigamobile.domain.repositories.IUsersRepository
+import io.eugenethedev.taigamobile.repositories.utils.TestCommonTask
+import io.eugenethedev.taigamobile.repositories.utils.getTestTasks
+import io.eugenethedev.taigamobile.testdata.TestData
+import kotlinx.coroutines.runBlocking
+import org.junit.Test
+import java.time.LocalDate
+import kotlin.test.*
+
+
+class TasksRepositoryTest : BaseRepositoryTest() {
+    lateinit var tasksRepository: ITasksRepository
+    lateinit var sprintsRepository: ISprintsRepository
+    lateinit var usersRepository: IUsersRepository
+
+    @BeforeTest
+    fun setupSprintsRepositoryTest() {
+        tasksRepository = TasksRepository(mockTaigaApi, mockSession)
+        sprintsRepository = SprintsRepository(mockTaigaApi, mockSession)
+        usersRepository = UsersRepository(mockTaigaApi, mockSession)
+    }
+
+    @Test
+    fun `test get working on`() = runBlocking {
+        val listCommonTasks = tasksRepository.getWorkingOn().sortedBy { it.title }
+
+        val totalTestCommonTasks = TestData.projects
+            .flatMapIndexed { index, project -> project.epics + getTestTasks(index) + project.issues + project.userstories }
+            .filter { it.assignedTo == TestData.activeUser && !it.isClosed }
+            .map {
+                TestCommonTask(
+                    title = it.title,
+                    assignee = it.assignedTo,
+                    isClosed = it.isClosed
+                )
+            }
+            .sortedBy { it.title }
+
+        listCommonTasks.forEachIndexed { index, data ->
```

Comment:
Sizes assert? Also, here and in a lot of places below you use the same code for asserting tasks equality. This significantly increases amount of similar code and makes everything bloated. Consider hoisting this part in the separate function (and do not forget sizes assert)